src/context.cpp

```diff
@@ -616,12 +639,44 @@ namespace librealsense
             if( option_id == RS2_OPTION_COUNT )
                 throw librealsense::invalid_value_exception( to_string() << "Option " << option->get_name() << " type not found" );
 
-            auto opt = std::make_shared< rs2_dds_option >( option );
+            auto opt = std::make_shared< rs2_dds_option >( option,
+                [&]( const std::string & name, float value ) { set_option( name, value ); },
+                [&]( const std::string & name ) -> float { return query_option( name ); } );
             register_option( option_id, opt );
         }
 
-        //float get_option( rs2_option option ) const override;
-        //void set_option( rs2_option option, float value ) const override;
+        void set_option( const std::string & name, float value ) const
+        {
+            // Sensor is setting the option for all supporting streams (with same value)
+            for( auto & stream : _streams )
+            {
+                for( auto & dds_opt : stream.second->options() )
+                {
+                    if( dds_opt->get_name().compare( name ) == 0 )
+                    {
+                        _dev->set_option_value( dds_opt, value );
+                        break;
+                    }
+                }
+            }
+        }
+
+        float query_option( const std::string & name ) const
+        {
+            for( auto & stream : _streams )
+            {
+                for( auto & dds_opt : stream.second->options() )
```

Comment:
maybe more efficient to use an unordered_map or something?

Reply:
Iterating over the whole `map` or `unordered_map` is both o(N).
But looking at other uses of `_streams` I see direct access to elements that should be a bit faster using `unordered_map` so I will change it.
We won't have that many streams (mostly single digit) so probably doesn't really matter...

Reply:
For `unordered_map` we need to define a hash function for `sid_index` struct. I will leave `map` for now, as I said it's a small container so won't noticeably effect performance